react-components/src/components/RuleBasedOutputs/RuleBasedOutputs.tsx

```diff
@@ -0,0 +1,75 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+import { useEffect, type ReactElement } from 'react';
+
+import { type CogniteCadModel } from '@cognite/reveal';
+import { useAllMappedEquipmentAssetMappings, useReveal } from '../..';
+import { Color } from 'three';
+import { type RuleOutputSet } from './types';
+import { generateRuleBasedOutputs } from './utils';
+
+export type ColorOverlayProps = {
+  ruleSet: RuleOutputSet | undefined;
+};
+
+export function RuleBasedOutputs({ ruleSet }: ColorOverlayProps): ReactElement | undefined {
+  const viewer = useReveal();
+
+  // only enabled whether has loaded models
+  if (viewer.models === undefined || viewer.models.length === 0) return;
+
+  const models = viewer.models;
+
+  const {
+    data: assetMappings,
+    isFetching,
+    hasNextPage,
+    fetchNextPage
+  } = useAllMappedEquipmentAssetMappings(models);
+
+  // clean up the appearance
+  models.forEach((model) => {
+    const currentModel = model as CogniteCadModel;
+    currentModel.removeAllStyledNodeCollections();
+
+    currentModel.setDefaultNodeAppearance({
+      color: new Color('#efefef')
+    });
+  });
+
+  useEffect(() => {
+    if (!isFetching && hasNextPage !== undefined) {
+      void fetchNextPage();
+    }
+  }, [isFetching, hasNextPage, fetchNextPage]);
+
+  useEffect(() => {
+    if (assetMappings === undefined || isFetching) return;
+    if (ruleSet === undefined) return;
+
+    const initializeRuleBasedOutputs = async (model: CogniteCadModel): Promise<void> => {
+      // parse assets and mappings
+      const flatAssetsMappingsList =
+        assetMappings?.pages
+          .flat()
+          .map((item) => item.mappings)
+          .flat() ?? [];
+      const flatMappings = flatAssetsMappingsList.map((node) => node.items).flat();
+      const contextualizedAssetNodes =
+        assetMappings?.pages
+          .flat()
+          .map((item) => item.assets)
+          .flat() ?? [];
+
```

Comment:
In  `assetMappings`, each of the pages contain one asset-mapping list per model, which means that only one of the mapping-lists per page is needed to consider here

Reply:
But should we also be prepared for multi models? To flat all in a one dimensional list to facilitate the logic?

Reply:
Isn't this happening within a loop where each iteration considers one model? So I thought we could just filter out the asset mappings relevant for the current iteration

Reply:
If I understood your suggestion correctly (only using the first page/per model), done here:
https://github.com/cognitedata/reveal/pull/4210/commits/03b40b181de31e95d5b302162f9a3d5d0b1ffb51

Reply:
reverted here: 
https://github.com/cognitedata/reveal/pull/4210/commits/c51354fb0dca063f92301efcff39f86fa9d9560b